pkg/proxy/proxier.go

```diff
@@ -328,6 +332,9 @@ func (proxier *Proxier) openPortal(service ServicePortName, info *serviceInfo) e
 			return err
 		}
 	}
+	if info.nodePort != 0 {
```

Comment:
We should keep a map[int]bool of all the ports (nodePorts and proxy ports, maybe) we have allocated and fail if there is a conflict.